src/imguiclient/imguiapp.cpp

```diff
@@ -2391,10 +2391,36 @@ void ImGuiApp::FileManagerDescriptorActions(ProgramFileDescriptor &descriptor, b
         // Change text color too to make it readable with the light ImGui color theme.
         ImScoped::StyleColor text_color1(ImGuiCol_Text, ImVec4(1.00f, 1.00f, 1.00f, 1.00f));
         ImScoped::StyleColor text_color2(ImGuiCol_TextDisabled, ImVec4(0.50f, 0.50f, 0.50f, 1.00f));
-        erased = ImGui::Button("Remove");
 
-        // #TODO: Guard this button press with a confirmation dialog ?
-        // There is an example for this in "Dear ImGui Demo" > "Popups & Modal windows" > "Modals".
+        if (ImGui::Button("Remove"))
+        {
+            ImGui::OpenPopup("Remove File?");
+        }
+
+        // Center the modal popup
+        ImVec2 center = ImGui::GetMainViewport()->GetCenter();
+        ImGui::SetNextWindowPos(center, ImGuiCond_Appearing, ImVec2(0.5f, 0.5f));
+
+        if (ImGui::BeginPopupModal("Remove File?", NULL, ImGuiWindowFlags_AlwaysAutoResize))
+        {
+            ImGui::Text("Are you sure you want to remove this file?\nThis operation cannot be undone!\n\n");
```

Comment:
I suggest say "Are you sure you want to remove this file from the list?\nIt does not delete it from disk.\n\n"